pkg/backend/filestate/store_test.go

```diff
@@ -395,3 +396,90 @@ func TestProjectReferenceStore_List(t *testing.T) {
 		})
 	}
 }
+
+func TestProjectReferenceStore_ProjectExists(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		desc string
+
+		// List of file paths relative to the storage root
+		// that should exist before ListReferences is called.
+		files []string
+
+		// Project name that should exist before ProjectExists is called.
+		projectName string
+
+		// Result that should be returned by ProjectExists.
+		exist bool
+
+		// Err that should be returned by ProjectExists.
+		err error
+	}{
+		{
+			desc:        "empty string project name",
+			files:       []string{},
+			projectName: "",
+			exist:       false,
+			err:         fmt.Errorf("empty project name"),
+		},
+		{
+			desc: "project exists",
+			files: []string{
+				".pulumi/stacks/a/foo.json",
+			},
+			projectName: "a",
+			exist:       true,
+			err:         nil,
+		},
+		{
+			desc: "project exists as empty directory",
```

Comment:
Hi @abhinav , I'm still not sure about this.

Is empty project considered as exist or not-exist?

Thanks!

Reply:
Considering this comment (see [comment](https://github.com/pulumi/pulumi/issues/12547#issuecomment-1528823478)), I assume that it is considered as not-exist.

LMK for concerns or anything.

Thanks!

Reply:
Yeah, this is correct. If the project doesn't have any stacks, then the project doesn't exist.